packages/near-fast-auth-signer/src/components/SignMultichain/utils.ts

```diff
@@ -0,0 +1,193 @@
+const FAST_AUTH_RELAYER_URL = 'http://34.136.82.88:3030';
+const CHAIN_CONFIG = {
+  ETH: {
+    providerUrl:
+      'https://sepolia.infura.io/v3/6df51ccaa17f4e078325b5050da5a2dd',
+  },
+  BNB: {
+    providerUrl: 'https://data-seed-prebsc-1-s1.bnbchain.org:8545',
+  },
+  BTC: {
+    networkType: 'testnet' as const,
+    // API ref: https://github.com/Blockstream/esplora/blob/master/API.md
+    providerUrl: 'https://blockstream.info/testnet/api/',
+  },
+};
+
+export const getMultiChainContract = () => (process.env.NETWORK_ID === 'mainnet' ? MULTICHAIN_CONTRACT_MAINNET : MULTICHAIN_CONTRACT_TESTNET);
+
+const getSchema = (asset: DerivationPathDeserialized['asset']) => {
+  switch (asset) {
+    case 'BTC':
+      return bitcoinSchema;
+    case 'ETH':
+    case 'BNB':
+      return evmSchema;
+    default:
+      return null;
+  }
+};
+
+export const validateMessage = async (message: MultichainInterface, asset: DerivationPathDeserialized['asset']): Promise<boolean
+| Error> => {
+  const schema = getSchema(asset);
+  if (!schema) {
+    return new Error(`Schema for asset ${asset} is not defined`);
+  }
+
+  try {
+    await schema.validate(message);
+    return true;
+  } catch (e) {
+    return new Error(e);
+  }
+};
+
+export const multichainAssetToCoinGeckoId = (asset: DerivationPathDeserialized['asset']) => {
+  const map = {
+    ETH:  'ethereum',
+    BNB:  'binancecoin',
+    BTC:  'bitcoin',
+  };
+
+  return map[asset] || null;
+};
+
+export const multichainAssetToNetworkName = (asset: DerivationPathDeserialized['asset']) => {
+  const map = {
+    ETH:  'Ethereum Network',
+    BNB:  'Binance Smart Chain',
+    BTC:  'Bitcoin Network',
+  };
+
+  return map[asset] || null;
+};
+
+export const getMultichainCoinGeckoPrice = async (asset: DerivationPathDeserialized['asset']) => fetchGeckoPrices(multichainAssetToCoinGeckoId(asset));
+
+const convertTokenToReadable = (value : MultichainInterface['value'], asset: DerivationPathDeserialized['asset']) => {
+  if (EVM_LIST.includes(asset)) {
+    return parseFloat(formatEther(value));
+  }
+  if (asset === 'BTC') {
+    return Bitcoin.toBTC(Number(value));
+  }
+  return Number(value);
+};
+
+export const getTokenAndTotalPrice = async (asset: DerivationPathDeserialized['asset'], value: MultichainInterface['value']) => {
+  const id = multichainAssetToCoinGeckoId(asset);
+  if (id) {
+    const res = await getMultichainCoinGeckoPrice(asset);
+    if (!res) {
+      return {
+        price:       0,
+        tokenAmount: convertTokenToReadable(value, asset)
+      };
+    }
+    const tokenPrice: number = res[id].usd;
+    const tokenAmount = convertTokenToReadable(value, asset);
+    return {
+      price: parseInt((tokenPrice * tokenAmount * 100).toString(), 10) / 100,
+      tokenAmount,
+      tokenPrice
+    };
+  }
+  return {
+    price:       0,
+    tokenAmount: convertTokenToReadable(value, asset)
+  };
+};
+
+export const shortenAddress = (address: string): string => {
```

Comment:
<img width="595" alt="Screenshot 2024-03-16 at 1 23 50 AM" src="https://github.com/near/fast-auth-signer/assets/22358004/7736d526-24d1-4746-bb28-a206a932137e">

There are some duplicated numbers, is this a bug? I think the user would expect the two parts of the string to be not overlapping. 

That's what I would expect